### PR TITLE
Resolve homedir references in Jedi environment path

### LINKED ISSUE #393: Environment path doesn't expand user directory
[Previously raised](https://github.com/davidhalter/jedi/issues/1941#issuecomment-1585150875) for Jedi project but deemed irrelevant at their level.

When using Virtualenvwrapper, or just for convenience keeping all the virtual Python environments in one directory, it's not possible to
to use user home directory references in `pylsp.plugins.jedi.environment`.  E.g. `~/.virtualenvs/myenv1`.

Neither _python-lsp-server_ nor Jedi don't expand `~`. 

This is a small thing but prevents effective sharing of dev environments between users (because the username would have to be explicitly hardcoded here).

I suggest using `os.path.expanduser` in this function:

https://github.com/python-lsp/python-lsp-server/blob/297099c6afdf86dfe9a9dd9f0736b32d91a65cdc/pylsp/workspace.py#L380-L401

### PR DESCRIPTION
**Pull request by odiroot**
Resolves: https://github.com/python-lsp/python-lsp-server/issues/393

Allows easier use with _virtualenvwrapper_ and portability of LSP configuration.

### CODE REVIEW

**Comment by odiroot** [comment]
@ccordoba12 I struggled to come up with a test for it.

I can use the `doc` fixture and go with 
```py
script = doc.jedi_script()
path = script._inference_state.environment.path
assert ...
```

But I have no idea how to inject my custom Jedi plugin configuration to that `doc`. It just follows my global settings.

File: pylsp/workspace.py [hunk 1]
```diff
@@ -389,6 +389,12 @@ def jedi_script(self, position=None, use_document_path=False):
             jedi.settings.auto_import_modules = jedi_settings.get('auto_import_modules',
                                                                   DEFAULT_AUTO_IMPORT_MODULES)
             environment_path = jedi_settings.get('environment')
+            # Jedi itself cannot deal with homedir-relative paths.
+            # On systems, where it is expected, expand the home directory.
+            import ipdb; ipdb.set_trace()
```
Comment (ccordoba12) [inline pylsp/workspace.py]:
```suggestion
```

I guess you used this line for testing, right?

**Review by ccordoba12 — COMMENTED** [review]
Small suggestion, the rest looks good to me.

Also, don't worry about adding a test for this. I think it's a very simple addition that doesn't merit a test.

File: pylsp/workspace.py (hunk 1 above)
Comment (odiroot) [reply pylsp/workspace.py]: Oh really sorry, indeed. I was trying to debug the tests. Pushed too hastily.

**Review by ccordoba12 — APPROVED** [review]
Thanks @odiroot!

**Pull request merged**